src/http_microhttpd.c

```diff
@@ -605,6 +677,18 @@ static const char *get_redirect_url(struct MHD_Connection *connection)
 	MHD_get_connection_values(connection, MHD_GET_ARGUMENT_KIND, &collect_query_key, &query_key);
 
 	if (!query_key.value)
+			return NULL;
+
+	return query_key.value;
+}
+
+static const char *get_voucher(struct MHD_Connection *connection)
+{
+	struct collect_query_key query_key = { .key = "voucher" };
+
+	MHD_get_connection_values(connection, MHD_GET_ARGUMENT_KIND, &collect_query_key, &query_key);
+
+	if (!query_key.value)
```

Comment:
There seem to be no point to test query_key.value for NULL if we return NULL anyway in that case.